### ISSUE TITLE
Conflicts with no-null-keyword

### ISSUE THREAD

**Issue by jupl** [palantir/tslint#2782]
### Bug Report

- __TSLint version__: 5.2.0
- __TypeScript version__: 2.3.0
- __Running TSLint via__: CLI

NOTE: Using `--type-check`.

#### TypeScript code being linted

```ts
export function test() {
  const el = document.querySelector('.value')
  if(el != undefined) {
    console.log(el.className)
  }
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-null-keyword": true,
    "strict-type-predicates": true,
    "triple-equals": [true, "allow-undefined-check"]
  }
}
```

#### Actual behavior
Lint error to use `=== null` instead, which can't be used due to `no-null-keyword`. Since `querySelector` could return `null` and `null !== undefined`, I can't do `=== undefined` either.

#### Expected behavior
No errors
